contrib/dhcp_agent/dhcp_agent.py

```diff
@@ -0,0 +1,399 @@
+             '[Default: all; Options: {}]'.format(', '.join(MODES)),
+    )
+    parser.add_option(
+        '-l',
+        '--log-path',
+        help='Path to log file. [Default: STDOUT]',
+        default='STDOUT',
+    )
+    parser.add_option(
+        '-c',
+        '--dhcp-config-entries',
+        help='Path to the DHCP entries configuration file.',
+    )
+    parser.add_option(
+        '-n',
+        '--dhcp-config-networks',
+        help='Path to the DHCP networks configuration file.',
+    )
+    parser.add_option(
+        '-a',
+        '--proto',
+        type='choice',
+        choices=PROTOS,
+        default=PROTO_HTTPS
+    )
+    parser.add_option(
+        '-e',
+        '--net-env',
+        help='Only get config for the specified network environment.',
+    )
+    parser.add_option(
+        '-d',
+        '--dc',
+        help='Only get config for the specified data center.',
+    )
+    parser.add_option(
+        '-r',
+        '--restart',
+        help='Restart service after fetch config?',
+        action='store_true',
+    )
+    parser.add_option(
+        '-v',
+        '--verbose',
+        help='Increase verbosity.',
+        action='store_true',
+        default=False,
+    )
+    parser.add_option(
+        '-s',
+        '--dhcp-service-name',
+        help='Name of the service to restart.',
+        default=DEFAULT_DHCP_SERVICE_NAME
+    )
+    return parser
+
+
+def _setup_logging(filename, verbose=False):
+    log_size = 20  # MB
+    logger = logging.getLogger(__file__)
+    if verbose:
+        logger.setLevel(logging.INFO)
+    else:
+        logger.setLevel(logging.WARNING)
+    if not filename or filename in ('-', 'STDOUT'):
+        handler = logging.StreamHandler()
+    else:
+        handler = logging_handlers.RotatingFileHandler(
+            filename, maxBytes=(log_size * (1 << 20)), backupCount=5
+        )
+    fmt = logging.Formatter("[%(asctime)-12s.%(msecs)03d] "
+                            "%(levelname)-8s %(filename)s:%(lineno)d  "
+                            "%(message)s", "%Y-%m-%d %H:%M:%S")
+    handler.setFormatter(fmt)
+
+    logger.addHandler(handler)
+    return logger
+
+
+def _remove_apication_lock(lockfile, logger):
+    logger.info('Remove lock')
+    os.unlink(lockfile)
+
+
+def _set_script_lock(logger):
+    lockfile = '{}.lock'.format(
+        os.path.join(tempfile.gettempdir(), os.path.split(sys.argv[0])[1])
+    )
+    f = os.open(lockfile, os.O_TRUNC | os.O_CREAT | os.O_RDWR)
+    try:
+        fcntl.lockf(f, fcntl.LOCK_EX | fcntl.LOCK_NB)
+        os.write(f, '{}'.format(os.getpid()).encode())
+        atexit.register(_remove_apication_lock, lockfile, logger)
+    except IOError as e:
+        if e.errno == errno.EAGAIN:
+            logger.critical('Script already running.')
+            sys.exit(2)
+        raise
+
+
+def _get_cmd_params_from_parser(parser):
+    """The wrapper to parser which returned dict instead of tuple.
```

Comment:
Do you mean something like "Wrapper around parser returning a dict instead of a tuple"..?